src/seedu/addressbook/commands/SortCommand.java

```diff
@@ -0,0 +1,38 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.*;
+
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Sorted the address book in lexicographical order.\n"
+            + "Example: " + COMMAND_WORD;
+
+    @Override
+    public CommandResult execute() {
+        List<ReadOnlyPerson> allPersons = sortPersonLists(addressBook.getAllPersons().immutableListView());
+        return new CommandResult(getMessageForPersonListSorted(allPersons), allPersons);
+    }
+
+    /**
+     * Sort the list in lexicographical order using names
+     *
+     *
+     * @return list of supported persons
+     */
+    private List<ReadOnlyPerson> sortPersonLists (List<ReadOnlyPerson> unsortedList){
+        List<ReadOnlyPerson> sortedListOfPersons = new ArrayList<ReadOnlyPerson>(unsortedList);
+        Collections.sort(sortedListOfPersons, new Comparator<ReadOnlyPerson>() {
```

Comment:
Why use an anonymous Comparator which compares name when you could just use the compareTo method you have overidden in person class(You could make ReadOnlyPerson extend Comparable)